src/Storages/StorageMergeTree.cpp

```diff
@@ -2666,21 +2672,25 @@ NameSet StorageMergeTree::MutationsSnapshot::getAllUpdatedColumns() const
 
 MergeTreeData::MutationsSnapshotPtr StorageMergeTree::getMutationsSnapshot(const IMutationsSnapshot::Params & params) const
 {
-    std::lock_guard lock(currently_processing_in_background_mutex);
+    MutationCounters mutations_snapshot_counters;
+    MutationsSnapshot::MutationsByVersion mutations_snapshot;
 
-    auto res = std::make_shared<MutationsSnapshot>(params, mutation_counters);
-    if (!res->hasAnyMutations())
-        return res;
+    std::lock_guard lock(currently_processing_in_background_mutex);
+    if (!params.need_data_mutations && !params.need_alter_mutations && mutation_counters.num_metadata <= 0)
```

Comment:
It is probably better completely remove this, because even this check is not 100% accurate, because `mutation_counters` can be changed between acquisition of a snapshot of parts and mutations. And we can't use approach from rmt because for regular mt mutation entry does not contain alter version